src/Sulu/Bundle/AdminBundle/Controller/AdminController.php

```diff
@@ -193,8 +194,19 @@ public function metadataAction(string $type, string $key, Request $request): Res
         $locale = $user->getLocale();
 
         $metadataOptions = $request->query->all();
-        $metadata = $this->metadataProviderRegistry->getMetadataProvider($type)
-            ->getMetadata($key, $locale, $metadataOptions);
+
+        try {
+            $metadata = $this->metadataProviderRegistry->getMetadataProvider($type)
+                ->getMetadata($key, $locale, $metadataOptions);
+        } catch (ServiceNotFoundException $exception) {
+            $view = View::create(
+                ['message' => \sprintf('"%s" not found', $exception->getSourceId())],
+                Response::HTTP_NOT_FOUND
+            );
+            $view->setFormat('json');
+
+            return $this->viewHandler->handle($view);
+        }
```

Comment:
unexpected change for this pull request:

```suggestion
        $metadata = $this->metadataProviderRegistry->getMetadataProvider($type)
            ->getMetadata($key, $locale, $metadataOptions);
```